### PR TITLE
Issue #1915 - Support _sort with _include and _revinclude

### PR DESCRIPTION
**Pull request by tbieste**
Issue #1915 - Support _sort with _include and _revinclude

Signed-off-by: Troy Biesterfeld <tbieste@us.ibm.com>

### CODE REVIEW

File: fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java [hunk 1]
```diff
@@ -801,9 +791,7 @@ public static FHIRSearchContext parseQueryParameters(Class<?> resourceType,
                 if (isSearchResultParameter(name)) {
                     parseSearchResultParameter(resourceType, context, name, params, lenient);
                     // _include and _revinclude parameters cannot be mixed with _summary=text
-                    // TODO: this will fire on each search result parameter; maybe move this above to where we handle _sort + _include/_revinclude?
-                    if (context.getSummaryParameter() != null
-                            && context.getSummaryParameter().equals(SummaryValueSet.TEXT)) {
+                    if (SummaryValueSet.TEXT.equals(context.getSummaryParameter())) {
```
Comment (michaelwschroeder) [inline fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java:794]: Seems like this check should be moved to the bottom after everything has been parsed, maybe before the check on line 925. Should also probably add a log message if we're removing parameters from the search.

File: fhir-search/src/test/java/com/ibm/fhir/search/test/InclusionParameterParseTest.java [hunk 2]
```diff
@@ -328,6 +348,34 @@ public void testIncludeValidTargetType() throws Exception {
...
+        assertEquals(sortParm.getType(), Type.STRING);
+        assertFalse(searchContext.hasIncludeParameters());
+        assertTrue(searchContext.hasRevIncludeParameters());
+        assertEquals(searchContext.getRevIncludeParameters().size(), 1);
+        assertEquals(searchContext.getRevIncludeParameters().get(0), revIncParm);
+
+        String selfUri = SearchUtil.buildSearchSelfUri("http://example.com/Patient", searchContext);
+        assertTrue(selfUri.contains(queryString));
+    }
+
```
Comment (michaelwschroeder) [inline fhir-search/src/test/java/com/ibm/fhir/search/test/InclusionParameterParseTest.java:378]: might want to add a test for _include/_revinclude with _summary=text

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java [hunk 3]
```diff
@@ -1369,28 +1367,27 @@ private FHIRPersistenceNotSupportedException buildNotSupportedException(String m
...
 
     /**
-     * This method takes the passed list of sorted Resource ids, acquires the Resource corresponding to each id, and returns those Resources in a List,
-     * sorted according to the input sorted ids.
-     * @param context - The FHIR persistence context for the current request.
+     * This method takes the passed list of sorted Resource ids, acquires the ResourceDTO corresponding to each id,
+     * and returns those ResourceDTOs in a List, sorted according to the input sorted ids.
      * @param resourceType - The type of Resource that each id in the passed list represents.
      * @param sortedIdList - A list of Resource ids representing the proper sort order for the list of Resources to be returned.
      * @param elements - An optional list of element names to include in the resources. If null, filtering will be skipped.
```
[the remark this answers is not in the document]
Comment (tbieste) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java:1374]: Fixed.

File: fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java (hunk 1 above)
Comment (tbieste) [reply fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java:794]: Moved to top. Treat as error if strict, but log message and handle as before if lenient.

File: fhir-search/src/test/java/com/ibm/fhir/search/test/InclusionParameterParseTest.java (hunk 2 above)
Comment (tbieste) [reply fhir-search/src/test/java/com/ibm/fhir/search/test/InclusionParameterParseTest.java:409]: Added.

**Review by michaelwschroeder — APPROVED** [review]
LGTM

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**